src/sage/schemes/projective/coherent_sheaf.py

```diff
@@ -0,0 +1,323 @@
+            sage: X = P2.subscheme(x^4 + y^4 + z^4)
+            sage: s = X.structure_sheaf(3)
+            sage: s.defining_twist()
+            3
+        """
+        return self._twist
+
+    def defining_module(self):
+        """
+        Return the module defining this coherent sheaf.
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme(x^4 + y^4 + z^4)
+            sage: s = X.structure_sheaf()
+            sage: s.defining_module()
+            Quotient module by Submodule of Ambient free module of rank 1 over the integral domain Quotient
+             of Multivariate Polynomial Ring in x, y, z over Rational Field by the ideal (x^4 + y^4 + z^4)
+             Generated by the rows of the matrix:
+              []
+        """
+        return self._module
+
+    def _cohomology_group(self, r=0):
+        """
+        Return the `r`-th cohomology group as a vector space.
+
+        The vector space is the internal representation of the cohomology group.
+
+        See :mod:`sage.schemes.projective.cohomology` to see how the internal
+        representation is computed.
+
+        INPUT:
+
+        - ``r`` -- (default: 0) a non-negative integer
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme(x^4 + y^4 + z^4)
+            sage: s = X.structure_sheaf()
+            sage: s._cohomology_group(1)
+            Vector space quotient V/W of dimension 3 over Rational Field where
+            V: Vector space of degree 3 and dimension 3 over Rational Field
+            Basis matrix:
+            [1 0 0]
+            [0 1 0]
+            [0 0 1]
+            W: Vector space of degree 3 and dimension 0 over Rational Field
+            Basis matrix:
+            []
+            sage: s._cohomology_group(1).dimension()  == s.cohomology(1)
+            True
+        """
+        return self._cohomology.H(r)
+
+    def cohomology(self, r=0):
+        """
+        Return the dimension of the `r`-th cohomology as a vector space.
+
+        INPUT:
+
+        - ``r`` -- (default: 0) a non-negative integer
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme(x^4 + y^4 + z^4)
+            sage: sheaf = X.structure_sheaf()
+            sage: sheaf.cohomology(0)
+            1
+            sage: sheaf.cohomology(1)
+            3
+            sage: sheaf.cohomology(2)
+            0
+        """
+        return self._cohomology.h(r)
+
+    def twist(self, t=0):
+        r"""
+        Return the twisted sheaf `F(n)` of this sheaf `F`.
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme(x^4 + y^4 + z^4)
+            sage: sh = X.structure_sheaf()
+            sage: sh.twist(1).euler_characteristic()
+            2
+            sage: sh.twist(2).euler_characteristic()
+            6
```

Comment:
Please add a doctest showing the twist of a twisted shift adds.

Reply:
Good idea. Done.